azure-iot-device/azure/iot/device/iothub/pipeline/iothub_pipeline.py

```diff
@@ -54,6 +54,8 @@ def __init__(self, auth_provider, pipeline_configuration):
             .append_stage(pipeline_stages_iothub.HandleTwinOperationsStage())
             .append_stage(pipeline_stages_base.CoordinateRequestAndResponseStage())
             .append_stage(pipeline_stages_iothub_mqtt.IoTHubMQTTConverterStage())
+            .append_stage(pipeline_stages_base.RetryStage())
+            .append_stage(pipeline_stages_base.TimeoutStage())
```

Comment:
So, for clarity. These timeouts and retries are only for MQTT operations at this point? We aren't doing higher level retries?

Reply:
Yes, it's in the comments in TimeoutStage and RetryStage. :)

---
In reply to: [337145290](https://github.com/Azure/azure-iot-sdk-python/pull/365#discussion_r337145290) [](ancestors = 337145290)